src/dropdowns/dropdown.ts

```diff
@@ -0,0 +1,209 @@
+  private clickEventSubscription: any;
+
+  @HostListener('keydown.esc', ['$event'])
+  @HostListener('keydown.tab')
+  onKeydownClose($event: Event) {
+    this.toggle(false);
+    if ($event) {
+      setTimeout(() => {
+        if (this.triggers.length) {
+          this.triggers.first.focusSelf();
+        }
+      });
+    }
+  }
+  @HostListener('keydown.arrowdown', ['$event'])
+  @HostListener('keydown.arrowright', ['$event'])
+  onKeydownFocusNext($event: Event) {
+    $event.preventDefault();
+    this.focusNextItem();
+  }
+  @HostListener('keydown.arrowup', ['$event'])
+  @HostListener('keydown.arrowleft', ['$event'])
+  onKeydownFocusPrevious($event: Event) {
+    $event.preventDefault();
+    this.focusPreviousItem();
+  }
+
+  constructor(private elementRef: ElementRef, private pageClickService: NglPageClickService) {}
+
+  ngOnInit() {
+    this.openEventSubscription = openEventEmitter.subscribe(this.handleDropdownOpenEvent.bind(this));
+    this.clickEventSubscription = this.pageClickService.subscribe(this.handlePageClickEvent.bind(this), () => <boolean>this.isOpen);
+  }
+
+  ngAfterContentInit() {
+    this._registerToTriggers();
+    this._registerToItems();
+    setTimeout(() => {
+      this._updateItems(this.values || []);
+    });
+  }
+
+  ngOnDestroy() {
+    this.openEventSubscription.unsubscribe();
+    this.clickEventSubscription.unsubscribe();
+  }
+
+  toggle(toggle?: boolean) {
+    toggle = toggle === undefined ? !this.isOpen : toggle;
+    if (toggle === this.isOpen) {
+      return;
+    }
+    this.isOpenChange.emit(toggle);
+    if (toggle) {
+      openEventEmitter.emit(this);
+      this.focusNextItem();
+    }
+  }
+
+  selectItem(value: any, isSelected: boolean) {
+    this.selectionEmitter.emit({value, isSelected});
+  }
+
+  handlePageClickEvent($event: any) {
+    if (!this.handlePageEvents || !this.handlePageClickEvents) {
+      return;
+    }
+    const ancestorPath = this.getAncestorPathFromEvent($event);
+    if (!ancestorPath.filter((elm: HTMLElement) => elm === this.elementRef.nativeElement).length) {
+      this.toggle(false);
+    }
+  }
+
+  private _registerToTriggers() {
+    this.triggers.forEach(trigger => trigger.registerParent(this));
+  }
+
+  private _registerToItems() {
+    this.items.forEach(item => {
+      item.registerParent(this);
+    });
+  }
+
+  private _updateItems(values: any[]) {
+    this.items.forEach(item => {
+      item.handleSelections(values);
+    });
+  }
+
+  private focusNextItem() {
+    if (!this.items.length) {
+      return;
+    }
+    let activeElementIndex = -1;
+    let idx = 0;
+    this.items.forEach(item => {
+      if (item.hasFocus()) {
+        activeElementIndex = idx;
+      }
+      idx++;
+    });
+    activeElementIndex++;
+    if (activeElementIndex === this.items.length) {
+      activeElementIndex = this.items.length - 1;
+    }
+    this.items.toArray()[activeElementIndex].focusSelf();
+  }
+
+  private focusPreviousItem() {
```

Comment:
I see many similarities with `focusNextItem`. Can't you combine to one taking a direction parameter?